### ISSUE TITLE
build android faill

### ISSUE THREAD

**Issue by yk19129515895** [simplex-chat/simplex-chat#1272]
Hi I tried to build android apk completely by the following command: 'cd simplex-chat/scripts/android && ./build-android.sh stable", but failed with the following message:
...
  inflating: /tmp/tmp.n77TSXjupj/simplex-chat/apps/android/app/src/main/cpp/libs/arm64-v8a/libsupport.so
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build failed with an exception.

* Where:
Settings file '/tmp/tmp.n77TSXjupj/simplex-chat/apps/android/settings.gradle' line: 8

* What went wrong:
A problem occurred evaluating settings 'android'.
> Could not find method dependencyResolutionManagement() for arguments [settings_i3rxekieihbfljnvcor9dfp$_run_closure1@a3c974b] on settings 'android' of type org.gradle.initialization.DefaultSettings.

* Try: 
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s

**Comment by epoberezkin** [comment]
You need to build libraries first that the android app depends on (libsimplex.so, libsupport.so, libcrypto.so), the source code is in this repo as well – you have to use nix for that.

**Comment by shumvgolove** [comment]
Hello there @yk19129515895.

You should specify commit in [`nix-android`](https://github.com/simplex-chat/simplex-chat/commits/nix-android) branch, in particular commits containing `Merge branch 'master' into nix-android`. For example, https://github.com/simplex-chat/simplex-chat/commit/cb8b0131efe0b6b9757a54dbf0a03bb8c9640158, so the final command becomes:

```sh
$ ./build-android.sh cb8b0131efe0b6b9757a54dbf0a03bb8c9640158
```

**Comment by yk19129515895** [comment]
> You need to build libraries first that the android app depends on (libsimplex.so, libsupport.so, libcrypto.so), the source code is in this repo as well – you have to use nix for that.

Yes I also want to build libsimplex.so etc. but till now I don't know how to do it since I am not familiar with nix and the only document (scripts/nix/README.md) is too simple.

**Comment by yk19129515895** [comment]
> Hello there @yk19129515895.
> 
> You should specify commit in [`nix-android`](https://github.com/simplex-chat/simplex-chat/commits/nix-android) branch, in particular commits containing `Merge branch 'master' into nix-android`. For example, [cb8b013](https://github.com/simplex-chat/simplex-chat/commit/cb8b0131efe0b6b9757a54dbf0a03bb8c9640158), so the final command becomes:
> 
> ```shell
> $ ./build-android.sh cb8b0131efe0b6b9757a54dbf0a03bb8c9640158
> ```

I found that the issue actually is caused by my too old gradle version (v4.4, default  in Ubuntu 20.04), after manual installation of latest gradle (v7.5.1), now the issue disappears.
However, I have the following new issue after I run './build-android.sh cb8b0131efe0b6b9757a54dbf0a03bb8c9640158':
HEAD is now at cb8b0131 Merge branch 'master' into nix-android
trace: To make project.plan-nix for haskell-project a fixed-output derivation but not materialized, set `plan-sha256` to the output of the 'calculateMaterializedSha' script in 'passt
hru'.  
trace: To materialize project.plan-nix for haskell-project entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.
trace: haskell-nix.haskellLib.cleanGit: /nix/store/9danlavp3pm7qg46zyhpv8i0ckl3xsi8-source does not seem to be a git repository,
assuming it is a clean checkout.
trace: WARNING: No materialized dummy-ghc-data for aarch64-unknown-linux-android-aarch64-unknown-linux-android-ghc-8.10.7-x86_64-linux.
trace: To make this a fixed-output derivation but not materialized, set `sha256` to the output of the 'calculateMaterializedSha' script in 'passthru'.
trace: To materialize this entirely, pass a writable path as the `materialized` argument and run the 'updateMaterialized' script in 'passthru'.
trace: WARNING: 8.8.2 is out of date, consider using 8.8.4.
error: NAR for '/nix/store/9k7npmwjgaijzrd1a2g3n2lzi88khaqr-aarch64-unknown-linux-android-ghc-8.10.7' fetched from 'https://cache.zw3rk.com' is incomplete
error: some substitutes for the outputs of derivation '/nix/store/a8h9fa4b69waxhzgqgarsz2qk8hhw5n6-aarch64-unknown-linux-android-ghc-8.10.7.drv' failed (usually happens due to networking issues); try '--fallback' to build derivation from source
error: 1 dependencies of derivation '/nix/store/4jsd8lsakm78ylk7a5d9qkh7s8x5dnnr-dummy-data-aarch64-unknown-linux-android-ghc-8.10.7.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wypd8w0ni3vvwqr98hphq6bf9nsxxpga-dummy-aarch64-unknown-linux-android-ghc-8.10.7.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vk80gzab21nl

**Comment by shumvgolove** [comment]
Hey @yk19129515895 

There were recent enchantments for `nix build` could you try with `build-android.sh cc2e74842e943d9a6b9d6a9b807a342e8f04b21d`?

**Comment by mkurniafebrianto** [comment]
how do I find the 'nix-branch'? could you please tell me the way to find it? I already try to find that branch but it doesn't exist anymore in the latest repo. I really appreciate it if you want to help, thank you.
